service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java

```diff
@@ -30,40 +32,53 @@ public class TerraformManager {
     private static final ObjectMapper objectMapper = new ObjectMapper();
     public static final String SYNC_HCL_FILENAME = "sync.tf";
     public static final String ENVIRONMENT_KEY = "environment";
+    public static final String EXECUTION_LOG_FILE = "commandExecution.log";
     private final TerraformLogProcessingService terraformLogProcessingService;
     private final TerraformProperties terraformProperties;
     private final TerraformClientFactory terraformClientFactory;
 
     public TerraformManager(TerraformLogProcessingService terraformLogProcessingService,
-                            TerraformProperties terraformProperties, TerraformClientFactory terraformClientFactory) {
+                            TerraformProperties terraformProperties,
+                            TerraformClientFactory terraformClientFactory) {
         this.terraformLogProcessingService = terraformLogProcessingService;
         this.terraformProperties = terraformProperties;
         this.terraformClientFactory = terraformClientFactory;
     }
 
-    public void execute(CommandRequest request, Command command, Map<String, String> envVars) {
-
+    public Path execute(CommandRequest request, Command command, Map<String, String> envVars) {
         MDC.put(RouteConstants.COMMAND_CORRELATION_ID, request.getCommandCorrelationId());
         MDC.put(RouteConstants.MESSAGING_SERVICE_ID, request.getServiceId());
-
         log.debug("Executing command {} for serviceId {} correlationId {} context {}", command.getCommand(), request.getServiceId(),
                 request.getCommandCorrelationId(), request.getContext());
-
         setEnvVarsFromParameters(command, envVars);
-
+        PrintWriter executionLogWriter = null;
+        Path executionLogFilePath = null;
         try (TerraformClient terraformClient = terraformClientFactory.createClient()) {
-
             Path configPath = TerraformUtils.createConfigPath(request, terraformProperties.getWorkingDirectoryRoot());
+            Path executionLogPath = TerraformUtils.createCommandExecutionLogDir(configPath);
+            executionLogFilePath = executionLogPath.resolve(command.getCommand() + "-" + EXECUTION_LOG_FILE);
+            executionLogWriter = new PrintWriter(new FileOutputStream(executionLogFilePath.toString(), false), true);
```

Comment:
If the same command is executed multiple times (for the same `app` + `messagingService`), the execution log file will be rewritten. This is done so that, in case of an error, we don't stream stale log messages to ep core